lib/osf-components/addon/components/resource/component.ts

```diff
@@ -0,0 +1,71 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import Intl from 'ember-intl/services/intl';
+
+import ResourceModel from 'ember-osf-web/models/resource';
+
+interface ResourceArgs {
+    resource: ResourceModel;
+}
+
+export default class ResourceComponent extends Component<ResourceArgs> {
+    @service intl!: Intl;
+
+    srcFinal?: string = '/assets/images/badges/data_small_gray.png';
+
+    constructor(owner: unknown, args: ResourceArgs) {
+        super(owner, args);
+    }
+
+    get getSrc() {
+        if (!this.args.resource) {
+            return 0;
+        }
+        const resourceType = this.args.resource.resourceType;
+        const isFinal = this.args.resource.finalized;
```

Comment:
This is not what the `finalized` flag is for. `Finalized` means that we show the resource at all, and `not finalized` means that it's currently being created but not yet submitted. Outside of the edit dialog box, we shouldn't even be aware that a non-`finalized` Resource exists.

Reply:
That is as anticipated then. Would you like to pair to discuss other alternatives?

Reply:
See my comment about using the `has<Resource>` attributes on the registration model. If that doesn't make sense, we'll chat.

Reply:
Yes please. I understand the boolean and helper function, but it would be good to make sure we understand the implementation in the left side nav.